test/serializers/meta_test.rb

```diff
@@ -67,11 +67,38 @@ def test_meta_is_not_used_on_arrays
         assert_equal expected, adapter.as_json
       end
 
+      def test_meta_is_present_on_arrays_with_root
+        serializer = ArraySerializer.new([@blog], meta: {total: 10}, meta_key: "haha_meta")
+        adapter = ActiveModel::Serializer::Adapter::Json.new(serializer, root: 'blog')
+        expected = {
+          'blog' => [{
+            id: 1,
+            name: "AMS Hints",
+            writer: {
+              id: 2,
+              name: "Steve"
+            },
+            articles: [{
+              id: 3,
+              title: "AMS",
+              body: nil
+            }]
+          }],
+          'haha_meta' => {
+            total: 10
+          }
+        }
+        assert_equal expected, adapter.as_json
+      end
+
       private
 
       def load_adapter(options)
-        serializer = AlternateBlogSerializer.new(@blog, options)
-        ActiveModel::Serializer::Adapter::Json.new(serializer)
+        adapter_opts, serializer_opts =
+          options.partition { |k, _| ActionController::Serialization::ADAPTER_OPTION_KEYS.include? k }.map { |h| Hash[h] }
+
```

Comment:
wow is this screaming for a refactor of https://github.com/rails-api/active_model_serializers/blob/35fb9de3104336fde07df093a4bd7765ac9ea0fa/lib/action_controller/serialization.rb#L37-L49 or what?  (not in this PR, just saying)

I remember reading through all these tests and code while trying to figure out how to test my serializers.  I ended up doing something like in [my comment in 878](https://github.com/rails-api/active_model_serializers/issues/878#issuecomment-96073030) which was a lot more work that you'd think.  Also, too bad minitest doesn't have shared examples, though I think [ActiveModel::Lint](https://github.com/rails/rails/blob/7ba24831d7f809d54e88607db8f7d21780efbdfb/guides/source/active_model_basics.md#lint-tests) might be a fine pattern.